banyan/src/index.rs

```diff
@@ -101,6 +101,14 @@ pub trait CompactSeq: DagCbor {
     fn is_empty(&self) -> bool {
         self.len() == 0
     }
+    /// Provide a size hint, which is used to calculate the overall size of the
+    /// node, which is used for caching purposes.
+    /// This function should be overridden, if the used item type contains any
+    /// heap allocated objects, otherwise the default implementation is a rough
+    /// estimate.
+    fn size_hint(&self) -> usize {
```

Comment:
I would not call this size_hint, since that method for iterators gives you the number of items, not the in mem size.

Reply:
`estimated_size`? Not sure what a good name is. Maybe size_hint is fine after all.

Reply:
I agree that this is a potential source of confusion, so I favor `estimated_size`